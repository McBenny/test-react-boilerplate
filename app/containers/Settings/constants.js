/*
 * SettingsConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const CHANGE_COMPETITION = 'boilerplate/Settings/CHANGE_COMPETITION';
export const CHANGE_ROUND = 'boilerplate/Settings/CHANGE_ROUND';
export const CHANGE_GENDER = 'boilerplate/Settings/CHANGE_GENDER';
export const CHANGE_TEAM_NAME = 'boilerplate/Settings/CHANGE_TEAM_NAME';
export const ADD_EMPTY_MEMBER = 'boilerplate/Settings/ADD_EMPTY_MEMBER';
export const CHANGE_MEMBER = 'boilerplate/Settings/CHANGE_MEMBER';
export const INIT_SETTINGS = 'boilerplate/Settings/INIT_SETTINGS';

export const EMPTY_MEMBER_DATA = {
    id: 1,
    reference: 0,
    name: '',
    goals: 0,
    yellowCards: 0,
    redCards: 0,
    blueCards: 0,
    suspensions: 0
};
export const OFFICIALS_REFERENCES = ['A', 'B', 'C', 'D'];
export const EMPTY_MEMBER = {
    players: {
        ...EMPTY_MEMBER_DATA
    },
    officials: {
        ...EMPTY_MEMBER_DATA,
        reference: OFFICIALS_REFERENCES[0]
    }
};

export const MAX_NUMBER = {
    players: 16,
    officials: 4,
    yellowCards: 1,
    redCards: 1,
    blueCards: 1,
    suspensions: 2,
    timeouts: 3
};

export const GENDERS = {
    men: 'Men',
    women: 'Women',
    mixed: 'Mixed'
};

export const UUID_PREFIX = 'uuid-';

export const PERSONS_TYPES = {
    players: 'players',
    officials: 'officials'
};
