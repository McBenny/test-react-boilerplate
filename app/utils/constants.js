export const RESTART_ON_REMOUNT = '@@saga-injector/restart-on-remount';
export const DAEMON = '@@saga-injector/daemon';
export const ONCE_TILL_UNMOUNT = '@@saga-injector/once-till-unmount';

export const CHANGE_TEAM_A_NAME = 'CHANGE_TEAM_A_NAME';
