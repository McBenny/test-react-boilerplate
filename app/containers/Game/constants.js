/*
 * GameConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

import { EMPTY_PLAYER } from '../Settings/constants';

export const SAVE_SETTINGS = 'boilerplate/Game/SAVE_SETTINGS';
export const HANDLE_GAME_STATUS = 'boilerplate/Game/HANDLE_GAME_STATUS';
export const ADD_EVENT = 'boilerplate/Game/ADD_EVENT';
export const ADD_GOAL = 'boilerplate/Game/ADD_GOAL';
export const ADD_YELLOW_CARD = 'boilerplate/Game/ADD_YELLOW_CARD';
export const ADD_RED_CARD = 'boilerplate/Game/ADD_RED_CARD';
export const ADD_BLUE_CARD = 'boilerplate/Game/ADD_BLUE_CARD';
export const ADD_SUSPENSION = 'boilerplate/Game/ADD_SUSPENSION';
export const ADD_TIMEOUT = 'boilerplate/Game/ADD_TIMEOUT';

export const EVENT_TYPES = {
    goal: 'goal',
    yellowCard: 'yellow card',
    redCard: 'red card',
    blueCard: 'blue card',
    suspension: '2 minutes',
    timeout: 'time out',
    gameStart: 'start game',
    gameEnd: 'end game',
    periodStart: 'start period',
    periodEnd: 'end period'
};

export const PERIODS = {
    0: 'warm-up',
    1: 'half-time 1',
    2: 'half-time 2',
    3: 'extra-time half-time 1',
    4: 'extra-time half-time 2',
    5: 'full-time'
};

export const UNKNOWN_PLAYER = {
    ...EMPTY_PLAYER,
    id: 0,
    playerName: '(unknown)'
};
