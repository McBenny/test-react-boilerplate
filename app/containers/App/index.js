/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from 'react';
import { Switch, Route } from 'react-router-dom';

import HomePage from '../HomePage/Loadable';
import Game from '../Game/Loadable';
import NotFoundPage from '../NotFoundPage/Loadable';

import '../../styles/core.scss';
import { URLS } from './constants';

export default function App() {
    return (
        <div>
            <Switch>
                <Route exact path={URLS.index} component={HomePage} />
                <Route exact path={URLS.game} component={Game} />
                <Route component={NotFoundPage} />
            </Switch>
        </div>
    );
}
