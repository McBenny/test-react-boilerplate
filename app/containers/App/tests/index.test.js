import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import App from '../index';
import { URLS } from '../constants';

const renderer = new ShallowRenderer();

describe('<App />', () => {
    it('should render and match the snapshot', () => {
        renderer.render(<App />);
        const renderedOutput = renderer.getRenderOutput();
        expect(renderedOutput).toMatchSnapshot();
    });
});

describe('<App /> - Constants', () => {
    it('should have 2 specific keys', () => {
        expect(URLS.game).toBeDefined();
        expect(URLS.index).toBeDefined();
    });
});
