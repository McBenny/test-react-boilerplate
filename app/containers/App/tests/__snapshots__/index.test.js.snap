// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<React.Fragment>
  <ForwardRef(WithStyles) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/game"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/score-sheet"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</React.Fragment>
`;
